autogen/logger/file_logger.py

```diff
@@ -89,7 +89,7 @@ def log_chat_completion(
         thread_id = threading.get_ident()
         source_name = None
         if isinstance(source, str):
-            source_name = source
+            source_name = getattr(source, "name", "unknown")
         else:
             source_name = source.name
```

Comment:
```suggestion
            source_name = source
        else:
            source_name = getattr(source, "name", "unknown")
```

Reply:
this looks good. @drewgillson, can you please update this and add tests as mentioned.